controllers/drcluster_controller_test.go

```diff
@@ -408,59 +444,52 @@ var _ = Describe("DRClusterController", func() {
 		When("an S3Profile fails listing", func() {
 			It("reports NOT validated with reason s3ListFailed", func() {
 				By("modifying a DRCluster with an invalid S3Profile that fails listing")
-				drcluster.Spec.S3ProfileName = s3Profiles[4].S3ProfileName
-				Expect(k8sClient.Update(context.TODO(), drcluster)).To(Succeed())
-				drclusterConditionExpectEventually(drcluster, false, metav1.ConditionFalse, Equal("s3ListFailed"), Ignore(),
+				latestdrc := updateDRClusterParameters(drcluster.Name, "", nil, s3Profiles[4].S3ProfileName)
```

Comment:
This helper is strange - both getting latest version, update it, and returning the updated version.

What if we had API like this:

     drc = &DRCluster{}
     
     // Update values
     drc.Spec.S3ProfileName = s3Profiles[4].S3ProfileName

     err = syncDRCluster(drc)
     Expect(err).NotTo(HaveOccurred())

After a sync, all drc fields match the stored DRCluster. The code will get the latest
version, update drc from storage and update storage from drc.

This avoids this kind of calls which are very hard to understand and error prone (e.g. passing arguments in the wrong order when they have the same type):

    updateDRClusterParameters(drcluster.Name, "", nil, s3Profiles[4].S3ProfileName)

Reply:
We must change all the fields between get and update. Otherwise we missed the point of retryOnConflict function

Reply:
Here is the explanation about RetryOnConflict https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict

Reply:
Assuming that we change all fields, and many fields are using zero value (e.g. nil or ""), accepting a DRCluster with the configuration we want is still better and more
clear.

    // Create zero DRCluster (strings initialized to "", pointers to nil)
    drc = &DRCluster{}
 
    // Update values as needed for the specific test
     drc.Spec.S3ProfileName = s3Profiles[4].S3ProfileName

    // Sync wanted state with actual state using retries
    err = syncDRCluster(drc)
    
    // Fail if all sync failed
    Expect(err).NotTo(HaveOccurred())